elasticdl/python/master/task_manager.py

```diff
@@ -523,23 +526,23 @@ def _get_average_task_completed_time(self):
             A dict: It contains containing the average time
             for TRAINING and EVALUATION tasks.
         """
-        average_task_completed_time = {}
+        max_task_completed_time = {}
 
         if len(self._task_completed_times[elasticdl_pb2.TRAINING]) < 20:
-            average_task_completed_time[elasticdl_pb2.TRAINING] = 300
-        else:
-            average_task_completed_time[
+            max_task_completed_time[
                 elasticdl_pb2.TRAINING
-            ] = statistics.mean(
+            ] = _TASK_TIMEOUT_THRESHOLD
+        else:
+            max_task_completed_time[elasticdl_pb2.TRAINING] = max(
```

Comment:
self._task_completed_times[elasticdl_pb2.TRAINING] contains the completed time for all the Training tasks. The array will be large if there is a lot of data in the source table and the max function will be slower and slower. If we only care the max time, only update the max value for each completed tasks instead of recording the time of each task.